experimental/ir/lower_options.go

```diff
@@ -252,6 +257,14 @@ func (r optionRef) resolve() {
 			}
 		}
 
+		if pc.IsFirst() && field.InternedFullName() == r.session.langIDs.MapEntry {
```

Comment:
Could this get tickled with an option name like `(map_entry.key)`, for a custom option named "map_entry"? I think maybe we need the extra check that this is not an extension name.

Also, it is also forbidden to set `uninterpreted_option` directly, and that check appears to be missing. And, I know you haven't implemented Editions yet, but this is a great place for a TODO that `features` also cannot be set unless the syntax is editions.

Reply:
It's not. I added a test though.